### PR TITLE
Remove aes_gcm module

### LINKED ISSUE #133: [parity-crypto] Remove dependency on ring if sensible
A bit more controversial than #124 is removing the dependency with `ring`.

Related discussion: https://github.com/paritytech/substrate/pull/2415

### PR DESCRIPTION
**Pull request by tomaka**
Based on top of #139 
Closes #133. 

Removes `aes_gcm` module.

### CODE REVIEW

**Comment by dvdplm** [comment]
I'm fine closing #133 after this, but would like a trace to remain about the rest of the work. Normally I'd reject something like this because it's a bit half-baked, but I suspect that you need this in Substrate urgently enough to just fork if we don't merge as-is.
I would greatly appreciate it if you could make a brain dump of what you know about ripping out `ring` entirely, i.e. the stuff left behind the `aead` feature, in a ticket either here or on Whisper in `parity-ethereum`.
Thanks!

**Comment by tomaka** [comment]
I don't need this urgently, unless the definition of "urgent" is less than 2 months.

I want to get out of ring because it doesn't compile for WASM, and I went for a feature gate because ring is the only crate that provides an implementation of GCM (with the exception of rust-crypto, which we also got rid of).

It's not that I put a feature gate because I was too lazy, it's that there's no other way at the moment except by writing our own GCM crate, which I'm not qualified to do.

File: .travis.yml [hunk 1]
```diff
@@ -22,3 +22,4 @@ script:
   - cd uint/ && cargo test --features=std,quickcheck --release && cd ..
   - cd plain_hasher/ && cargo test --no-default-features && cd ..
   - cd parity-util-mem/ && cargo test --features=estimate-heapsize && cd ..
+  - cd parity-crypto/ && cargo test --all-features && cd ..
```
Comment (ordian) [inline .travis.yml]: I'm not sure how this works. Error conversion from ring [was removed](https://github.com/paritytech/parity-common/pull/144/files#diff-c6b3ae839850070966d556586872155dL107), but it's used e.g. [here](https://github.com/paritytech/parity-common/pull/144/files#diff-89cede331f3b34fa96c6102348e03a44R34) if I'm not mistaken. 
Also running this command locally doesn't seem to run aes_gcm tests.

**Comment by dvdplm** [comment]
> I don't need this urgently, unless the definition of "urgent" is less than 2 months.

Oh, ok, sorry got the wrong impression then.

How about we extract the GCM stuff into its own crate (or even, if Whisper is the only thing using it, maybe it can be just a module there)? That way we can move forward with `parity-crypto` being truly shared code.

File: .travis.yml (hunk 1 above)
Comment (tomaka) [reply .travis.yml]: Oh, interesting.

**Comment by tomaka** [comment]
So I had written `features = "aead"` instead of `feature = "aead"`, which means it was always disabled.

I put back a `SymmetricCrypto` variant in the error, but that doesn't expose any type from ring (ring's error is an opaque type that doesn't provide any information anyway).

**Comment by ordian** [comment]
I took a look at reverse dependencies of `parity-crypto`, 
[on crates.io](https://crates.io/crates/parity-crypto/reverse_dependencies) GCM is not used, for crates not published to crates.io, I've only searched for github and (apart from parity-ethereum copy-paste) found only parity-ethereum, substrate and https://github.com/Conflux-Chain/conflux-rust/blob/99a899bf129d66373890c2c5a3fe0e01938801a4/core/Cargo.toml#L36 usages, GCM is only used for whisper. Of course it doesn't take into account not published crates with private source, on gitlab, etc. But since we're making a breaking change, I would go for removing GCM from parity-crypto and putting it into whisper or its own crate as David suggested.

**Comment by tomaka** [comment]
Pushed another commit that removes `aes_gcm` altogether.

File: .travis.yml (hunk 1 above)
Comment (ordian) [reply .travis.yml]: This line is now not needed.

**Pull request merged**